src/services/external/excalidraw-backend/utils/handlers.ts

```diff
@@ -57,6 +58,7 @@ export const authorizeWithRoomAndJoinHandler = async (
     );
   }
 
+  socket.data.roomId = roomID;
```

Comment:
technically the socket can join multiple rooms. To not depend on the room joined, I would suggest to publish the event once in the broadcast listener, or feel free to try something else - I am just cautious to depend always on a single room.